src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DisplayMetadata.cs

```diff
@@ -40,10 +41,10 @@ public class DisplayMetadata
         public string DisplayFormatString { get; set; }
 
         /// <summary>
-        /// Gets or sets a display name for the model.
+        /// Gets or sets a delegate used for getting the display name for the model.
         /// See <see cref="ModelMetadata.DisplayName"/>
         /// </summary>
-        public string DisplayName { get; set; }
+        public Func<string, string> DisplayName { get; set; }
```

Comment:
If a user wishes to use a string key for a resource lookup directly, they can do that without enabling it at this level. They just need to create a lambda with a closure over the property name. This is important because the current approach forces _every_ user to create such a closure -- and most (as shown in your test changes) don't need the property name.

``` c#
  public class MyDisplayMetadataProvider
  {
    public void GetDisplayMetadata(DisplayMetadataProviderContext context)
    {
      if (context.Key.MetadataKind == ModelMetadataKind.Property)
      {
        context.DisplayMetadata.DisplayName = () => GetDisplayName(context.Key.Name);
      }
    }

    private static string GetDisplayName(string propertyName)
    {
      // Do something magical with propertyName.
      return propertyName;
    }
  }
```